src/System.Management.Automation/engine/parser/CharTraits.cs

```diff
@@ -299,15 +324,10 @@ internal static bool IsHexDigit(this char c)
         }
 
         // Return true if the character is a decimal digit.
-        internal static bool IsDecimalDigit(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.Digit) != 0;
-            }
+        internal static bool IsDecimalDigit(this char c) => (uint)(c - '0') <= 9;
 
-            return false;
-        }
+        // Return true if the character is a binary digit.
+        internal static bool IsBinaryDigit(this char c) => (uint)(c - '0') <= 1;
```

Comment:
While this is true, it seems more straightforward to just have `=> c == '0' || c == '1';`. It took me a minute or so to realise why this isn't a problem.

Reply:
(Perhaps the coercion + comparison has a performance advantage?)

Reply:
@rjmholt I was following @iSazonov's lead on that one, he apparently pulled that from the CoreFX code somewhere or other. Knowing nothing about how these checks would be performed on a hardware level means I can only speculate on potential ways this might be better than the ultimately far more readable version you propose.

Should I change it?

Reply:
@rjmholt The micro optimization come from CoreFX - 2 vs 3 operations - this gives huge perf win.

Reply:
Should I add a comment to this effect?

Reply:
Because we get the question the comment can help code readers.